proposals/lip-0053.md

```diff
@@ -738,14 +755,14 @@ The Lisk consensus mechanism is designed to create and publish certificates regu
 Setup to gather the required mainchain information:
 
 * Have access to the Interoperability module endpoints exposed by a mainchain node.
-* Maintain a list of all CCMs included in the sidechain outbox. For each height where a CCM was included in the outbox, also save the inclusion witness of the outbox root with respect to the state root. All CCMs and witnesses for heights that have been certified on the sidechain can be discarded.
-* Maintain a history of all validator changes on the mainchain for rounds that have not yet been certified on the sidechain.
+* Maintain a list of all CCMs included in the sidechain outbox. For each height where a CCM was included in the outbox, also save the inclusion witness of the outbox root with respect to the state root. CCMs can be discarded once they are included via a CCU in a final block of the sidechain. All witnesses corresponding to outbox root values older than the latest `partnerChainOutboxRoot` in the final part of the sidechain can be discarded as well.
+* Maintain a history of all validator changes on the mainchain for rounds that have not yet been certified by a certificate that is included in the the final part of the sidechain.
 
 Create a cross-chain update transaction for a given height `h1`:
 
-* Find a [signed certificate][lip-0061#select-certificate] in the mainchain block headers for a height (say `h2`) higher or equal to `h1`. This will be used as the `certificate` property of the transaction.
-* The property `inboxUpdate.crossChainMessages` lists a subset of the CCMs that have been included in the sidechain outbox up to `h2` (and which have not been included on the sidechain yet). If all CCMs are included, the `messageWitnessHashes` will be empty.
-* Compute the inclusion proof for the outbox root of the sidechain account with respect to the mainchain state root. This proof is then used to compute `inboxUpdate.outboxRootWitness`.
+* Find a [signed certificate][lip-0061#select-certificate] in the mainchain block headers for height `h1`. This will be used as the `certificate` property of the transaction.
```

Comment:
You may not necessarily find a certificate for every height. That's why it was formulated like this using `h2`.

Reply:
I am still confused as to why that would be a `cross-chain update transaction for a given height `h1`` rather than the cross-chain update transaction for `h2`.

Reply:
It describes the case where you start with a certain height value `h1` without caring if there exits indeed a certificate for this height. I would say this makes sense for the case when you would like to transmit CCMs created by a cross chain transaction included in height `h1`.

Reply:
I see, so the intro sentence `Create a cross-chain update transaction for a given height `h1`:` refers to creating a CCU that contains the messages up to `h1`.

Reply:
I partially reverted it back while incorporating your suggestion. I still think the inclusion proof should be the one for height `h2`

Reply:
> I still think the inclusion proof should be the one for height h2

I agree. Did I suggest something else? I'm sorry if I did.

Reply:
ah, I indeed suggested something wrong. Sorry.

Reply:
I resolve the comment